assets/src/edit-story/components/canvas/karma/cloneSelection.karma.js

```diff
@@ -0,0 +1,262 @@
+    return elements.length;
+  }
+  async function getElementByIndex(index) {
+    const {
+      state: {
+        currentPage: { elements },
+      },
+    } = await fixture.renderHook(() => useStory());
+    return elements[index - 1];
+  }
+  async function clickElement(elementId, isMultiSelect = false) {
+    const wrapper = await getCanvasElementWrapperById(elementId);
+    const rect = wrapper.getBoundingClientRect();
+    if (isMultiSelect) {
+      // hold shift
+      await fixture.events.keyboard.down('shift');
+    }
+    await fixture.events.mouse.click(rect.left + 1, rect.top + 1);
+    if (isMultiSelect) {
+      // release shift
+      await fixture.events.keyboard.up('shift');
+    }
+  }
+  async function clickAboveElement(elementId) {
+    const wrapper = await getCanvasElementWrapperById(elementId);
+    const rect = wrapper.getBoundingClientRect();
+    await fixture.events.mouse.click(rect.left, rect.top - 10);
+  }
+
+  // Low-level helpers
+  function getElementByQueryAndMatcher(tagName, matcher) {
+    return Array.from(fixture.querySelectorAll(tagName)).find(matcher);
+  }
+
+  function getByAttribute(attr, value) {
+    return (el) =>
+      typeof value === 'string'
+        ? el.getAttribute(attr) === value
+        : value.test(el.getAttribute(attr));
+  }
+
+  function getMediaElement(imageAlt) {
+    return getElementByQueryAndMatcher(
+      '[data-testid="mediaElement"] img',
+      getByAttribute('alt', imageAlt)
+    );
+  }
+
+  function getCanvasElementWrapperById(id) {
+    return fixture.querySelector(
+      `[data-testid="safezone"] [data-element-id="${id}"]`
+    );
+  }
+});
+
+const customMatchers = {
+  toBeEmpty: () => ({
+    compare: function (actual) {
+      const innerHTML = actual?.innerHTML ?? '';
+      const pass = innerHTML === '';
+      return {
+        pass,
+        message: pass
+          ? `Expected element to not be empty`
+          : `Expected element to be empty`,
+      };
+    },
+  }),
+  toHaveStyle: (util, customEqualityTesters) => ({
+    compare: function (element, property, expected) {
+      const actual = getComputedStyle(element)[property];
+      const pass = util.equals(actual, expected, customEqualityTesters);
+      return {
+        pass,
+        message: pass
+          ? `Expected element to not have style "${property}: ${expected}"`
+          : `Expected element to have style "${property}: ${expected}" but found "${actual}"`,
+      };
+    },
+  }),
+  toHaveProperty: (util, customEqualityTesters) => ({
+    compare: function (element, property, expected) {
+      const actual = element?.[property] ?? '';
+      const pass =
+        typeof expected === 'string'
+          ? util.equals(actual, expected, customEqualityTesters)
+          : expected.test(actual);
+      return {
+        pass,
+        message: pass
+          ? `Expected element to not have src "${expected}"`
+          : `Expected element to have src "${expected}" but found "${actual}"`,
```

Comment:
^ Is using "src" intended here, for the general `toHaveProperty`? Is guess in this context it's OK since it's always the src you're comparing?

Reply:
Fixed in 091fad534.